apis/cloudidentity/v1beta1/group_types.go

```diff
@@ -0,0 +1,148 @@
+
+	For Google-managed entities, the id must be the email address of an existing
+	group or user.
+
+	For external-identity-mapped entities, the id must be a string conforming
+	to the Identity Source's requirements.
+
+	Must be unique within a namespace. */
+	// +required
+	Id string `json:"id"`
+
+	/* Immutable. The namespace in which the entity exists.
+
+	If not specified, the EntityKey represents a Google-managed entity
+	such as a Google user or a Google Group.
+
+	If specified, the EntityKey represents an external-identity-mapped group.
+	The namespace must correspond to an identity source created in Admin Console
+	and must be in the form of 'identitysources/{identity_source_id}'. */
+	Namespace *string `json:"namespace,omitempty"`
+}
+
+type CloudIdentityGroupSpec struct {
+	/* An extended description to help users determine the purpose of a Group.
+	Must not be longer than 4,096 characters. */
+	Description *string `json:"description,omitempty"`
+
+	/* The display name of the Group. */
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="GroupKey field is immutable"
+	/* Immutable. EntityKey of the Group. */
+	// +required
+	GroupKey GroupGroupKey `json:"groupKey"`
+
+	/* Immutable. The initial configuration options for creating a Group.
+
+	See the
+	[API reference](https://cloud.google.com/identity/docs/reference/rest/v1beta1/groups/create#initialgroupconfig)
+	for possible values. Default value: "EMPTY" Possible values: ["INITIAL_GROUP_CONFIG_UNSPECIFIED", "WITH_INITIAL_OWNER", "EMPTY"]. */
+	InitialGroupConfig *string `json:"initialGroupConfig,omitempty"`
+
+	/* One or more label entries that apply to the Group. Currently supported labels contain a key with an empty value.
+
+	Google Groups are the default type of group and have a label with a key of cloudidentity.googleapis.com/groups.discussion_forum and an empty value.
+
+	Existing Google Groups can have an additional label with a key of cloudidentity.googleapis.com/groups.security and an empty value added to them. This is an immutable change and the security label cannot be removed once added.
+
+	Dynamic groups have a label with a key of cloudidentity.googleapis.com/groups.dynamic.
+
+	Identity-mapped groups for Cloud Search have a label with a key of system/groups/external and an empty value. */
+	// +required
+	Labels map[string]string `json:"labels"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Parent field is immutable"
+	/* Immutable. The resource name of the entity under which this Group resides in the
+	Cloud Identity resource hierarchy.
+
+	Must be of the form identitysources/{identity_source_id} for external-identity-mapped
+	groups or customers/{customer_id} for Google Groups. */
+	// +required
+	Parent string `json:"parent"`
+
+	/* Immutable. The service-generated name of the resource. Used for acquisition only. Leave unset to create a new resource. */
```

Comment:
suggest explain that this is a service-generated ID so the field can only be used for acquisition (this is different than other TF based resources IIRC)